base_location/models/company.py

```diff
@@ -6,19 +6,46 @@
 
 
 class ResCompany(models.Model):
-
     _inherit = 'res.company'
 
-    @api.onchange('better_zip_id')
-    def on_change_city(self):
-        if self.better_zip_id:
-            self.zip = self.better_zip_id.name
-            self.city = self.better_zip_id.city
-            self.state_id = self.better_zip_id.state_id
-            self.country_id = self.better_zip_id.country_id
+    city_id = fields.Many2one(
+        'res.city',
+        compute='_compute_address',
+        inverse='_inverse_city_id',
+        string="City"
+    )
 
-    better_zip_id = fields.Many2one(
+    zip_id = fields.Many2one(
         'res.better.zip',
-        string='Location',
+        string='ZIP Location',
+        compute='_compute_address',
+        inverse='_inverse_zip_id',
+        oldname="better_zip_id",
         help='Use the city name or the zip code to search the location',
     )
+
+    def _get_company_address_fields(self, partner):
```

Comment:
Explain what this method does.

Reply:
This comment still remains.

Reply:
This method computes address fields from the partner. Used in `_compute_address` as shown in [company_code](https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_company.py#L106)